platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/PlatformInitializationException.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.system;
+
+/**
+ * Thrown when an issue occurs while initializing/re-initializing platform instances
+ */
+public class PlatformInitializationException extends RuntimeException {
+    public PlatformInitializationException(final String message, final Throwable cause) {
+        super(message, cause);
+    }
+}
```

Comment:
I know the "conventional wisdom" is to always create specific exceptions for things. My personal take is that having a large number of exceptions that you don't intend to catch and handle doesn't really buy much. In this situation, I think a naked `RuntimeException` or `IllegalStateException` would be sufficient.

Reply:
I actually agree that we either use one of the naked exceptions or use one of the closest exceptions we have defined e.g.`PlatformConstructionException` - which is what I initially used but @lpetrovic05 has a different view, so I created this one.

Reply:
https://github.com/hashgraph/hedera-services/pull/12377#discussion_r1543003765